ontology/yaml/resources/HVAC/entity_types/VAV.yaml

```diff
@@ -463,4 +463,15 @@ VAV_RD_CO2C:
   implements:
   - VAV
   - RD
-  - CO2C
\ No newline at end of file
+  - CO2C
+
+VAV_SD_SDM_CO2C_SRC:
+  description: "CO2 controlled VAV, with damper, flowrate and scheduled run timer"
+  is_canonical: true
+  implements:
+  - VAV
+  - SD
+  - SDM
```

Comment:
Is there a separate supply damper that returns just a status and command? The combination of `SD` and `SDM` imply that all of these fields exist on the same device:
- `supply_air_damper_percentage_command`
- `supply_air_damper_command`
- `supply_air_damper_status`

If you have two separate supply dampers, then we would need to distinguish them from each other (`supply_air_damper_percentage_command` and `supply_air_isolation_damper_command`); if they are the same damper, how can it have both a percentage command and a binary command? We shouldn't have two commands for the same thing.

Reply:
i see. thanks sir. i will check the device again.

Reply:
we have documents in the repository which could be helpful. if you haven't read them already, please check these out:

- https://github.com/google/digitalbuildings/blob/master/ontology/docs/faq.md
- https://github.com/google/digitalbuildings/blob/master/ontology/docs/hvac_ahu.md
- https://github.com/google/digitalbuildings/blob/master/ontology/docs/hvac_fcu.md
- https://github.com/google/digitalbuildings/blob/master/ontology/docs/model_hvac.md

we are also working on some training modules which we will publish in the near future, but the docs above are your best resource for your current task.

so far your proposals have been good. I feel like you have a good understanding of the basic points (e.g. create abstract types when there is some functionality we want to reuse). most of my comments are more nuanced, and you will simply need some more time with the ontology before it becomes clearer to you. hopefully those docs above also clarify a few additional things, such as which fields we care about.

Reply:
> Is there a separate supply damper that returns just a status and command? The combination of `SD` and `SDM` imply that all of these fields exist on the same device:
> 
> * `supply_air_damper_percentage_command`
> * `supply_air_damper_command`
> * `supply_air_damper_status`
> 
> If you have two separate supply dampers, then we would need to distinguish them from each other (`supply_air_damper_percentage_command` and `supply_air_isolation_damper_command`); if they are the same damper, how can it have both a percentage command and a binary command? We shouldn't have two commands for the same thing.

For this sir, you are correct, I intend to apply a MISSING in the points later (supply_air_damper_percentage_command: MISSING). Because the SD and SDM combo is the nearest one I could find among all the others. I felt this is easiest to do on my part since I have minimal changes...but is that correct? Should I redo it like below and get only the parts I need? (Note: this is almost the same question earlier for FAN_SS_AL)

```
VAV_SDM_CO2C_SRC_HLFSS:
  description: "CO2 controlled VAV, with damper, hi/low flowrate setpoint and sensor and schedule run"
  is_canonical: true
  implements:
  - VAV
  - SDM
  - CO2C
  - SRC
  uses:
  - low_limit_supply_air_flowrate_setpoint
  - high_limit_supply_air_flowrate_setpoint
  - supply_air_flowrate_sensor 
```

Reply:
I am still unclear what this device does and what data it will provide. It seems like it is a variable-position air damper, with a flow setpoint that can vary between the low- and high-limits. It controls CO2 levels by varying the flowrate setpoint between those limits.

If this is the case, I would expect these fields:
- supply_air_damper_percentage_command
- supply_air_flowrate_sensor
- supply_air_flowrate_setpoint
- low_limit_supply_air_flowrate_setpoint
- high_limit_supply_air_flowrate_setpoint
- zone_air_co2_concentration_sensor
- zone_air_co2_concentration_setpoint
- (plus the schedule fields for SRC)

If this is the case, i would expect it to look like this:
```
implements:
- VAV
- SD
- CO2C
- SRC
uses:
- low_limit_supply_air_flowrate_setpoint
- high_limit_supply_air_flowrate_setpoint
```

You could also create a new abstract type to handle the bounding of the supply_air_flowrate_setpoint between the boundaries like this:
```
HLFC:
  description: "High and low limit bounded supply air flowrate setpoint control."
  uses:
  - low_limit_supply_air_flowrate_setpoint
  - high_limit_supply_air_flowrate_setpoint
  - supply_air_flowrate_setpoint
```

Reply:
Thanks sir, the final one is much more simplified now:

```
VAV_SDM_CO2C_SRC
  description: "CO2 controlled VAV, with damper, hi/low flowrate sensor and setpoint and schedule run"
  is_canonical: true
  implements:
  - VAV
  - SDM
  - CO2C
  - SRC
  uses:
  - low_limit_supply_air_flowrate_setpoint
  - high_limit_supply_air_flowrate_setpoint
  - supply_air_flowrate_sensor 
```
Thanks

Reply:
I am still confused on control logic. You have used `SDM`, which is two-position control. How does it control to a flow setpoint with two-position control on the damper? And how do the low and high limits relate to the target setpoint, and where is the target setpoint?

Reply:
I see what you mean. I'll ask again from my colleagues.

Reply:
You are correct sir on SDM. Will this be ok?

```
VAV_CO2C_SRC:
  description: "CO2 controlled VAV, with damper, hi/low flowrate sensor and setpoint and schedule run"
  is_canonical: true
  implements:
  - VAV
  - CO2C
  - SRC
  uses:
  - low_limit_supply_air_flowrate_setpoint
  - high_limit_supply_air_flowrate_setpoint
  - supply_air_flowrate_sensor 
  - supply_air_damper_command
  - supply_air_damper_percentage_sensor
```

Reply:
in this case, wouldnt it simplify even further to `VAV_SD_HLFC_CO2C_SRC`? you also still need the flowrate setpoint, which i assume you have because it is bounded by the high and low limits.

again, perfectly acceptable to have `MISSING` fields in your translation if the data is not available.

Reply:
understood sir. I see your point about MISSING and I did not look into SD and HLFC further hehe. Thanks again.

Reply:
@josephedwardchang what is the action here? i still expect that this be adjusted based on the comments above.

Reply:
yes sir. that's what I came up with. the HLSAFS contains the needed points. and I added 2 sensors for waht we need: co2 and flowrate sensors.

Reply:
i am still confused on the function here. i don't understand how you can have a two-position damper with flowrate setpoints... it won't modulate to control airflow to a specific value, it will just be whatever value it is based on geometry of the box and back pressure in the duct. does the box use two-position damper control (as indicated by the description) or does it use modulating, continuous control (that is, will it have `supply_air_damper_command` or `supply_air_damper_percentage_command`)?

 also, for co2, why not implement CO2C?

Reply:
Hi again sir,

This is the list of points for the VAV we need:
```
CO2 Setpoint		
Minimum Air Flow Rate Setpoint		
Maximum Air Flow Rate Setpoint		
Air Flow Rate Present Value		
VAV Damper Control	Modulating Damper	%
VAV Damper Feedback	Modulating Damper	%
Time Schedule
```
I may have been wrong on the dampers as they are clearly % (sorry for that), does the list seem valid? I have no idea if it is, as CO2 is stated as setpoint but doesn't contain CO2 sensor. With your last comments I'm leaning towards this now:

```
VAV_PDSCV_CO2C_HLSAFS_SRC:
  description: "VAV with CO2 monitor, dampers, hi/low flowrate setpoint and sensor, schedule run"
  is_canonical: true
  implements:
  - VAV
  - PDSCV
  - CO2C
  - HLSAFS
  - SRC
  uses:
  - supply_air_flowrate_sensor
```
Thanks

Reply:
it is missing its supply_air_flowrate_setpoint? i cannot have setpoint boundaries without the septoint itself. i also think it needs to use SD instead of PDSCV.

Reply:
you're right sir, I didn't see this combination before hehe. Thanks, will do this.